persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java

```diff
@@ -0,0 +1,118 @@
+package com.baeldung.jdbcmetadata;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class MetadataExtractor {
+	private final DatabaseMetaData databaseMetaData;
+
+	public MetadataExtractor(Connection connection) throws SQLException {
+		this.databaseMetaData = connection.getMetaData();
```

Comment:
This line is using tabs. Be sure to use the standard code formatter (available in the tutorials repo, under eclipse or intelliJ foder, depending on which IDE you're using)